### PR TITLE
add term methods, small fixes, added jsdocs, use nyc

### PR DESCRIPTION
**Pull request by bergos**

### CODE REVIEW

File: test/Clownface/literal.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+  it('should use the given number as Literal', () => {
+    const value = 123
+    const cf = clownface(graph)
+
+    const result = cf.literal(value)
+
+    assert.strictEqual(result._context.length, 1)
+    assert.strictEqual(result._context[0].term.termType, 'Literal')
+    assert.strictEqual(result._context[0].term.value, value.toString())
```
Comment (tpluscode) [inline test/Clownface/literal.js]: Should also expect a numeric datatype here?

File: lib/term.js [hunk 2]
```diff
@@ -1,42 +1,55 @@
 const rdf = require('@rdfjs/data-model')
 
-function term (value, type, datatype) {
-  if (datatype) {
-    datatype = rdf.namedNode(datatype.value || datatype.toString())
+function blankNode (value) {
+  if (value && typeof value !== 'string') {
+    throw new Error('unknown type')
```
Comment (tpluscode) [inline lib/term.js]: This error is a bit vague

```suggestion
    throw new Error('Blank node identifier must be a string')
```

File: lib/term.js [hunk 3]
```diff
@@ -1,42 +1,55 @@
...
   }
 
-  if (type) {
-    if (type === 'BlankNode') {
-      return rdf.blankNode(value && value.toString())
-    }
+  return rdf.blankNode(value && value.toString())
```
Comment (tpluscode) [inline lib/term.js]: Isn't `value.toString()` unnecessary if you test the `typeof value`?
```suggestion
  return rdf.blankNode(value)
```

File: lib/term.js [hunk 4]
```diff
@@ -1,42 +1,55 @@
...
+}
+
+function literal (value, languageOrDatatype) {
+  if (typeof value !== 'string' && typeof value !== 'number') {
+    throw new Error('unknown type')
```
Comment (tpluscode) [inline lib/term.js]: Ditto on vague error.
```suggestion
    throw new Error('The value cannot be converted to a literal node')
```

File: lib/term.js [hunk 5]
```diff
@@ -1,42 +1,55 @@
...
 
-  if (type) {
-    if (type === 'BlankNode') {
-      return rdf.blankNode(value && value.toString())
-    }
+  return rdf.blankNode(value && value.toString())
+}
+
+function literal (value, languageOrDatatype) {
+  if (typeof value !== 'string' && typeof value !== 'number') {
```
Comment (tpluscode) [inline lib/term.js]: Boolean is also a valid literal.

```suggestion
  if (typeof value !== 'string' && typeof value !== 'number' && typeof value !== 'boolean') {
```

File: lib/term.js [hunk 6]
```diff
@@ -1,42 +1,55 @@
...
   }
 
-  if (!value) {
-    return undefined
+  return rdf.literal(value.toString(), languageOrDatatype)
+}
+
+function namedNode (value) {
+  if (typeof value !== 'string') {
+    throw new Error('unknown type')
```
Comment (tpluscode) [inline lib/term.js]:
```suggestion
    throw new Error('Named node must be a URI string')
```

Besides, would you consider checking that the value is indeed a URI? Maybe simply "substringing" for a colon?

File: lib/term.js [hunk 7]
```diff
@@ -1,42 +1,55 @@
...
   }
 
-  // TODO: library for JavaScript type -> xsd
+  if (type === 'BlankNode') {
+    return blankNode(value)
+  }
+
+  if (!value) {
+    return undefined
+  }
```
Comment (tpluscode) [inline lib/term.js]: Should you check for truthiness before returning `blankNode()`?

**Comment by bergos** [comment]
Separate PRs were generate for this PR.

**Pull request closed without merging**